src/os/bluestore/BlueStore.cc

```diff
@@ -3926,6 +3926,7 @@ void BlueStore::MempoolThread::_resize_shards(bool interval_stats)
 
   for (auto i : store->onode_cache_shards) {
     i->set_max(max_shard_onodes);
+    i->trim();
   }
   for (auto i : store->buffer_cache_shards) {
     i->set_max(max_shard_buffer);
```

Comment:
should we do the same for buffer_cache_shards?

Reply:
It is also missing for buffers.
Whether we should copy this solution depends on possibility to have truly running cluster that does not read/write. Or possibility to have some Buffer shards that for some reasons are dormant.

Reply:
I don't think we need this for buffers unless the size of a given buffer can change behind the buffer cache's back (which it shouldn't).  We exactly track how many bytes are in the cache and know how big a cached buffer is.

The onode case is special because caching an onode can result in a variable amount of required memory that can change behind the cache's back depending on the number of extents/blobs/etc.  That's why the onode cache trims based on the number of items instead of the number of bytes, and we have the whole wonky periodic average metadata size calculation that calls set_max to change the maximum number of onodes to cache.